indexing-service/src/main/java/io/druid/indexing/common/task/ArchiveTask.java

```diff
@@ -91,12 +98,15 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
       log.info("OK to archive segment: %s", unusedSegment.getIdentifier());
     }
 
-    // Move segments
-    for (DataSegment segment : unusedSegments) {
-      final DataSegment archivedSegment = toolbox.getDataSegmentArchiver().archive(segment);
-      toolbox.getTaskActionClient().submit(new SegmentMetadataUpdateAction(ImmutableSet.of(archivedSegment)));
+    if (toolbox.getTaskActionClient().submit(new LockUpgradeAction(getInterval()))) {
+      // Move segments
+      for (DataSegment segment : unusedSegments) {
+        final DataSegment archivedSegment = toolbox.getDataSegmentArchiver().archive(segment);
```

Comment:
sounds good...when you say release and reacquire lock - are you saying release the entire lock, reacquire it and then upgrade it to be non-preemptible ? or just downgrade the lock to be preemptible and upgrade it again ?